File: PlaybyPlayEFCore10/FSharp/myapi/Migrations/WeatherContextModelSnapshot.fs
namespace MyAPI.Migrations

open System
open Microsoft.EntityFrameworkCore
open Microsoft.EntityFrameworkCore.Infrastructure
open Microsoft.EntityFrameworkCore.Metadata
open Microsoft.EntityFrameworkCore.Migrations
open MyAPI.DataAccess

[<DbContext(typeof<WeatherContext>)>]
type WeatherContextModelSnapshot() =
    inherit ModelSnapshot()

    override this.BuildModel(modelBuilder: ModelBuilder) =
        modelBuilder
            .HasAnnotation("ProductVersion", "1.0.1")
            .HasAnnotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn)
            |> ignore

        modelBuilder.Entity("MyAPI.Domain.Comment", 
            fun b ->
                b.Property<int>("CommentID").ValueGeneratedOnAdd() |> ignore 
                b.Property<int>("ReactionId") |> ignore
                b.Property<string>("Text") |> ignore
                b.HasKey("CommentID") |> ignore
                b.HasIndex("ReactionId") |> ignore
                b.ToTable("Comments") |> ignore
                
            )|> ignore

        modelBuilder.Entity("MyAPI.Domain.Reaction", 
            fun b ->
                b.Property<int>("Id").ValueGeneratedOnAdd() |> ignore
                b.Property<string>("Name") |> ignore
                b.Property<string>("Quote") |> ignore
                b.Property<int>("WeatherEventId") |> ignore
                b.HasKey("Id") |> ignore
                b.HasIndex("WeatherEventId") |> ignore
                b.ToTable("Reactions") |> ignore) |> ignore

        modelBuilder.Entity("MyAPI.Domain.WeatherEvent",
            fun b ->
                b.Property<int>("Id").ValueGeneratedOnAdd() |> ignore
                b.Property<DateTime>("Date") |> ignore
                b.Property<string>("MostCommonWord") |> ignore
                b.Property<TimeSpan>("Time") |> ignore
                b.Property<int>("Type") |> ignore
                b.HasKey("Id") |> ignore
                b.ToTable("WeatherEvents") |> ignore) |> ignore

        modelBuilder.Entity("MyAPI.Domain.Comment", 
            fun b ->
                b.HasOne("MyAPI.Domain.Reaction")
                 .WithMany("Comments")
                 .HasForeignKey("ReactionId") |> ignore) |> ignore

        modelBuilder.Entity("MyAPI.Domain.Reaction",
            fun b ->
                b.HasOne("MyAPI.Domain.WeatherEvent")
                 .WithMany("Reactions")
                 .HasForeignKey("WeatherEventId")
                 .OnDelete(DeleteBehavior.Cascade) |> ignore) |> ignore

File: PlaybyPlayEFCore10/CSharp/myapi/Migrations/WeatherContextModelSnapshot.cs
﻿using System;
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Infrastructure;
using Microsoft.EntityFrameworkCore.Metadata;
using Microsoft.EntityFrameworkCore.Migrations;
using EFCoreWebAPI;

namespace myapi.Migrations
{
    [DbContext(typeof(WeatherContext))]
    partial class WeatherContextModelSnapshot : ModelSnapshot
    {
        protected override void BuildModel(ModelBuilder modelBuilder)
        {
            modelBuilder
                .HasAnnotation("ProductVersion", "1.0.0-rtm-21431");

            modelBuilder.Entity("EFCoreWebAPI.Comment", b =>
                {
                    b.Property<int>("Id")
                        .ValueGeneratedOnAdd();

                    b.Property<int?>("ReactionId");

                    b.Property<string>("Text");

                    b.HasKey("Id");

                    b.HasIndex("ReactionId");

                    b.ToTable("Comments");
                });

            modelBuilder.Entity("EFCoreWebAPI.Reaction", b =>
                {
                    b.Property<int>("Id")
                        .ValueGeneratedOnAdd();

                    b.Property<string>("Name");

                    b.Property<string>("Quote");

                    b.Property<int>("WeatherEventId");

                    b.HasKey("Id");

                    b.HasIndex("WeatherEventId");

                    b.ToTable("Reactions");
                });

            modelBuilder.Entity("EFCoreWebAPI.WeatherEvent", b =>
                {
                    b.Property<int>("Id")
                        .ValueGeneratedOnAdd();

                    b.Property<DateTime>("Date");

                    b.Property<string>("MostCommonWord");

                    b.Property<TimeSpan>("Time");

                    b.Property<int>("Type");

                    b.HasKey("Id");

                    b.ToTable("WeatherEvents");
                });

            modelBuilder.Entity("EFCoreWebAPI.Comment", b =>
                {
                    b.HasOne("EFCoreWebAPI.Reaction")
                        .WithMany("Comments")
                        .HasForeignKey("ReactionId");
                });

            modelBuilder.Entity("EFCoreWebAPI.Reaction", b =>
                {
                    b.HasOne("EFCoreWebAPI.WeatherEvent")
                        .WithMany("Reactions")
                        .HasForeignKey("WeatherEventId")
                        .OnDelete(DeleteBehavior.Cascade);
                });
        }
    }
}
